hazelcast-client/src/test/java/com/hazelcast/client/connectionstrategy/ConfiguredBehaviourTest.java

```diff
@@ -104,12 +104,22 @@ public void testReconnectModeOFFSingleMember() {
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.getConnectionStrategyConfig().setReconnectMode(OFF);
         HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+        final CountDownLatch disconnectedLatch = new CountDownLatch(1);
+        client.getLifecycleService().addLifecycleListener(new LifecycleListener() {
+            @Override
+            public void stateChanged(LifecycleEvent event) {
+                if (LifecycleEvent.LifecycleState.SHUTDOWN.equals(event.getState())) {
```

Comment:
It should be enough to wait for CLIENT_DISCONNECTED, right? why use SHUTDOWN state?